cpp/include/cudf/detail/null_mask.hpp

```diff
@@ -268,9 +268,9 @@ std::pair<rmm::device_buffer, size_type> bitmask_or(
  * @param mask_size_bits The number of bits to be ANDed in each mask
  * @param stream CUDA stream used for device memory operations and kernel launches
  * @param mr Device memory resource used to allocate the returned device_buffer
- * @return rmm::device_buffer Output bitmask
+ * @return size_type Count of set bits
  */
-void inplace_bitmask_and(
+size_type inplace_bitmask_and(
```

Comment:
To match the rest of the file this should be cudf::size_type

Reply:
I was matching this: https://github.com/rapidsai/cudf/blob/7fa509d5715da71df125fe0508cf9c5cbac0a0c7/cpp/include/cudf/detail/null_mask.hpp#L234

The argument types also use `size_type`. How should we reconcile this?

Reply:
Personally I think we should explicitly use `cudf::size_type` instead of deducing the type based on namespace lookup.

Reply:
I'm okay with that in principle. We currently have significant variation on this pattern within the codebase, and even within individual files. I would like to limit the scope of this PR to the functions in the changeset, if you approve. Perhaps file an issue to discuss? (Is this a job for clang-tidy?)

Reply:
> I would like to limit the scope of this PR to the functions in the changeset, if you approve

Absolutely 

> Perhaps file an issue to discuss? (Is this a job for clang-tidy?)

I don't see an existing clang-tidy check for this